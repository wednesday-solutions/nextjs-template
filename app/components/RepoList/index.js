/**
 *
 * RepoList
 *
 */

import React from 'react';
import get from 'lodash/get';
import { Skeleton } from '@mui/material';
import PropTypes from 'prop-types';
import { useRouter } from 'next/router';
import T from '@components/Text';
import { CustomCard, CardContainer } from '../styled/repos';
import If from '../If/index';

const RepoList = (props) => {
  const { reposData, loading, repoName } = props;
  const router = useRouter();

  const renderSkeleton = () => {
    return (
      <>
        <Skeleton data-testid="skeleton" animation="wave" variant="text" height={40} />
        <Skeleton data-testid="skeleton" animation="wave" variant="text" height={40} />
        <Skeleton data-testid="skeleton" animation="wave" variant="text" height={40} />
      </>
    );
  };

  const items = get(reposData, 'items', []);
  const totalCount = get(reposData, 'totalCount', 0);
  const BlockText = (props) => <T display="block" {...props} />;
  return (
    <If condition={items.length !== 0 || loading} otherwise={renderSkeleton()}>
      <CardContainer data-testid="repo-list">
        {repoName && <BlockText id="search_query" values={{ repoName }} />}
        {totalCount !== 0 && <BlockText id="matching_repos" values={{ totalCount }} />}
        {items.map((item, index) => (
          <CustomCard
            variant="outlined"
            key={index}
            onClick={() => router.push(`/info/${item?.name}?owner=${item?.owner.login}`)}
          >
            <BlockText id="repository_name" values={{ name: item.name }} />
            <BlockText id="repository_full_name" values={{ fullName: item.fullName }} />
            <BlockText id="repository_stars" values={{ stars: item.stargazersCount }} />
          </CustomCard>
        ))}
      </CardContainer>
    </If>
  );
};

RepoList.propTypes = {
  loading: PropTypes.bool.isRequired,
  reposData: PropTypes.shape({
    totalCount: PropTypes.number,
    incompleteResults: PropTypes.bool,
    items: PropTypes.array
  }),
  repoName: PropTypes.string
};

export default RepoList;
