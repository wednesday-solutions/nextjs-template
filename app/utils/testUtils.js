import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import configureStore from '../configureStore';
import colors from '@themes/colors';
import { translationMessages, DEFAULT_LOCALE } from '../i18n';

export const getComponentStyles = (Component, props = {}) => {
  renderProvider(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);

  return window.getComputedStyle(componentRoots[0])._values;
};

export const renderProvider = (children) => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} key={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <ThemeProvider theme={colors}>{children}</ThemeProvider>
    </IntlProvider>
  );
};
export const timeout = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export const renderStoreProvider = (children) => {
  const store = configureStore({});
  return render(
    <Provider store={store}>
      <IntlProvider locale={DEFAULT_LOCALE} key={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
        <ThemeProvider theme={colors}>{children}</ThemeProvider>
      </IntlProvider>
    </Provider>
  );
};
