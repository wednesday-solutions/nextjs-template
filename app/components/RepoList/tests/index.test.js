/**
 *
 * Tests for RepoList
 *
 */

import React from 'react'
// import { fireEvent } from '@testing-library/dom'
import { renderProvider } from '@utils/testUtils'
import RepoList from '../index'

describe('<RepoList />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<RepoList loading />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should contain 1 RepoList component', () => {
    const { getAllByTestId } = renderProvider(<RepoList loading />)
    expect(getAllByTestId('repo-list').length).toBe(1)
  })
})
