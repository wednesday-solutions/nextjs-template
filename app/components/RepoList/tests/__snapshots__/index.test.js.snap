// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoList /> should render and match the snapshot 1`] = `
.emotion-0.emotion-0 {
  margin: 1.25rem 0;
  padding: 1.25rem;
  border: 1px solid rgba(0, 0, 0, 0.12);
  border-radius: 4px;
}

<body>
  <div>
    <div
      class="emotion-0"
      data-testid="repo-list"
    />
  </div>
</body>
`;
